keras_nlp/layers/transformer_decoder.py

```diff
@@ -208,6 +211,8 @@ def call(
         decoder_attention_mask=None,
         encoder_padding_mask=None,
         encoder_attention_mask=None,
+        cache=None,
```

Comment:
we need to document these and the extra output, especially as TransformerEncoder/TransformerDecoder are two of the main offerings from our layers.

Reply:
Still need to document return type!